core/src/OpenACC/Kokkos_OpenACCSpace.hpp

```diff
@@ -62,11 +69,16 @@ class OpenACCSpace {
   static constexpr char const* name() { return "OpenACCSpace"; }
 
  private:
-  void* impl_allocate(const Kokkos::Experimental::OpenACC& exec_space,
-                      const char* arg_label, const size_t arg_alloc_size,
+  template <typename ExecutionSpace>
+  void* impl_allocate(const ExecutionSpace& exec_space, const char* arg_label,
+                      const size_t arg_alloc_size,
                       const size_t arg_logical_size = 0,
-                      const Kokkos::Tools::SpaceHandle =
-                          Kokkos::Tools::make_space_handle(name())) const;
+                      const Kokkos::Tools::SpaceHandle arg_handle =
+                          Kokkos::Tools::make_space_handle(name())) const {
+    (void)exec_space;
```

Comment:
Slight preference for me, to be consistent with the other memory space allocate functions, we would have in the function declaration `void* impl_allocate(const ExecutionSpace&, ...)` instead of having `(void)exec_space`.

Reply:
Are you suggesting to separate the function template declaration from the function template definition?
If so, where should we put the function template definition? (We cannot put it in the `Kokkos_OpenACCSpace.cpp` since the definition will not be visible to the template instantiation points.)
If not suggesting separation, can you elaborate your suggestion?